docs/user/accumulo/commandline.rst

```diff
@@ -21,22 +21,21 @@ here are Accumulo-specific.
 General Arguments
 -----------------
 
-Most commands require you to specify the connection to Accumulo. This generally includes a username and
-password (or Kerberos keytab file). Specify the username and password with ``--user`` and ``--password``
-(or ``-u`` and ``-p``). In order to avoid plaintext passwords in the bash history and process list,
-the password argument may be omitted, in which case it will be prompted for instead.
-
-To use Kerberos authentication instead of a password, use ``--keytab`` with a path to a Kerberos keytab
-file containing an entry for the specified user. Since a keytab file allows authentication without any
-further constraints, it should be protected appropriately.
-
-If the necessary environment variables are set (generally as part of the install process), the tools should
-connect automatically to the Accumulo instance. To specify the connection instead, use ``--instance-name``
-and ``--zookeepers`` (or ``-i`` and ``-z``).
-
-The ``--auths`` and ``--visibilities`` arguments correspond to the ``AccumuloDataStore`` parameters
-``geomesa.security.auths`` and ``geomesa.security.visibilities``, respectively. See :ref:`authorizations`
-and :ref:`accumulo_visibilities` for more information.
+Most commands require you to specify the connection to Accumulo. This generally includes the instance name,
+zookeeper hosts, username, and password (or Kerberos keytab file). Specify the instance with ``--instance-name``
+and ``--zookeepers``, and the username and password with ``--user`` and ``--password``. The password argument may be
+omitted in order to avoid plaintext credentials in the bash history and process list - in this case it will be
+prompted case for later. To use Kerberos authentication instead of a password, use ``--keytab`` with a path to a
+Kerberos keytab file containing an entry for the specified user. Since a keytab file allows authentication
+without any further constraints, it should be protected appropriately.
+
+Instead of specifying the cluster connection explicitly, an appropriate ``accumulo-client.properties`` (for Accumulo
+2) or ``client.conf`` (for Accumulo 1) may be added to the classpath. See the
+`Accumulo documentation <https://accumulo.apache.org/docs/2.x/getting-started/clients#creating-an-accumulo-client>`_
+for information on the necessary configuration keys.
```

Comment:
Can we be explicit that once the accumulo-client.properties is on the classpath that the user does not need to specify all those parameters above?

Also, what happens if they mix and match?

Separately, is there a good way to use one set of command line tools to connect to multiple clusters?  (For that, I could imagine having a --accumulo-config option being helpful.

Just random thoughts...

Reply:
I thought it was pretty explicit, especially if you read the linked accumulo docs... i can make it more so.
data store params are first precedence and the props file is second.
i originally had logic to load the props from various places that accumulo checks, but decided that was a bit opaque... instead i modeled it after the hbase-site.xml logic. if you want to connect to multiple clusters, you can specify the connection directly, so i don't think we need to support an accumulo-config option.